plasmapy/classes/sources/coils.py

```diff
@@ -0,0 +1,166 @@
+from typing import Iterable
+
+import astropy.units as u
+import numpy as np
+
+from plasmapy.classes.plasma_base import GenericPlasma
+from plasmapy.formulary import magnetostatics
+
+E_unit = u.V / u.m
+
+
+class Coils(GenericPlasma):
+    def __init__(self, *magnetostatics: magnetostatics.CircularWire):
+        """
+        Work-in-progress class for vacuum magnetic fields due to MagnetoStatics`.
+
+        This is primarily helpful for `plasmapy.simulation.ParticleTracker`.
+
+        Note: currently this accepts only circular coils as magnetostatics.
+        Look for TODO in the code for the reason.
+
+        Parameters
+        ----------
+        magnetostatics : plasmapy.formulary.magnetostatics.CircularWire
+            any number of CircularWire
+        """
+        self.magnetostatics = magnetostatics
+
+    @staticmethod
+    def _interpolate_E(r: np.ndarray):
+        return np.zeros(r.shape)
+
+    def interpolate_E(self, r: u.m):
+        """
+        Returns the electric field, with units.
+
+        Note that MagnetoStatics provide no electric field, so this is effectively zero.
+
+        Parameters
+        ----------
+        r : u.m
+            Position; can be a (3,) array or a (N, 3) array.
+
+        Returns
+        -------
+        electric field array of the same shape as the input.
+        """
+        return u.Quantity(self._interpolate_E(r.si.value), E_unit)
+
+    def _interpolate_B(self, r: np.ndarray):
+        B = np.zeros(r.shape, dtype=float)
+        for ms in self.magnetostatics:
+            # TODO - I had trouble calling the function otherwise...
+            field = ms._magnetic_field(r, ms.pt, ms.dl, ms.current, ms.w)
+            B += field
+        return B
+
+    def interpolate_B(self, r: u.m):
+        """
+        Returns the magnetic field from all the MagnetoStatics, with units.
+
+        Parameters
+        ----------
+        r : u.m
+            Position; can be a (3,) array or a (N, 3) array.
+
+        Returns
+        -------
+        Magnetic field array of the same shape as the input.
+        """
+        return u.Quantity(self._interpolate_B(r.si.value), u.T, copy=False)
+
+    def __repr__(self):
+        names = []
+        for ms in self.magnetostatics:
+            name = f"{ms.__class__.__name__}({ms.current * u.A})"
+            names.append(name)
+        from collections import Counter
+
+        items = []
+        for key, value in Counter(names).items():
+            items.append(f"{value} x {key}")
+        return f"{self.__class__.__name__}({', '.join(items)})"
+
+    @classmethod
+    def is_datasource_for(cls, **kwargs):
+        match = "interpolate_B" in kwargs.keys()
+        return match
+
+    def visualize(self, figure=None):  # coverage: ignore
```

Comment:
```suggestion
    def visualize(self, figure=None):
```
My preference would be to keep counting the lines for visualization methods when checking test coverage.  Our coverage percentage will be lower, but it's useful to be able to find where we have untested code.  

An alternative would be to do a regression test and make sure that the hash of the plot object stays the same.  I think this is a strategy that SunPy has used.  The problem is that this seems to be a fragile test, since the hash could change due to upstream changes.  I'm not really sure on the best way to deal with testing plots...

Reply:
I initially did it like this as I had issues getting PyVista to run in CI environments. I've since figured out some tricks to do that.

I'll remove the coverage ignores, make this thread a separate issue, and then in follow-up PR I'll figure out a way to CI plot output (I've been meaning to figure that out anyway) and get this coverage back up as possible. Does that sound good?